src/ledger/LedgerTxn.cpp

```diff
@@ -1749,22 +1948,64 @@ LedgerTxnRoot::Impl::getNewestVersion(LedgerKey const& key) const
         return getFromEntryCache(cacheKey);
```

Comment:
I'm really not sure about having two layers of caches, I know @MonsieurNicolas was in favor of this originally but I don't remember why. The way I see it, only `LedgerTxnRoot` can add things to either cache. Since both caches are in complete control of `LedgerTxnRoot`, they should probably be consolidated into one cache. Whether the cache needs to be stored in the `Prefetcher` may be an overreach of responsibilities. As it stands now it is possible that something in the prefetch queue is also in the entry cache, leading to unnecessary loads. Perhaps the `Prefetcher` should just have read-only access to a cache stored somewhere else? Thoughts?

Reply:
This is now resolved in favor of having a single cache of large enough size.